src/include/sof/common.h

```diff
@@ -14,19 +14,24 @@
 /* Align the number to the nearest alignment value */
 #define IS_ALIGNED(size, alignment) ((size) % (alignment) == 0)
 
-#define is_non0_power_of_2(x) ((x) && !((x) & ((x) - 1)))
+/* Treat zero as a special case because it wraps around */
+#define is_power_of_2(x) ((x) && !((x) & ((x) - 1)))
 
-#define compile_fail_zero_or_true(x) (sizeof(struct{int _f : 1 - !(x); }))
+#define compile_assert(x) (sizeof(struct{int _f : 1 - !(x); }))
```

Comment:
it's a bit more than just an assertion. If `x == 0` compilation fails, that's correct. But if `x != 0` it doesn't just become a NOP, it also returns a non-0 value, which makes it usable in expressions like `compile_fail_zero_or_true(x) ? :` which are still verified at compile-time as long as `x` is a constant. So, not sure `compile_assert()` describes it adequately?

Reply:
Thanks @lyakh , indeed neither `assert()` nor `_Static_assert()` are expressions.

How about `sof_compile_test()` of `sof_compile_check()`?

Reply:
those sound better, yes. `if (compile_check(x == 0))` is probably also somewhat intuitive

Reply:
Renamed to `compile_check()`.

Also changed `compile_check() ? ALIGN() : 0 ` to `compile_check() ? ALIGN() : 0xBADCAFE ` to make it clearer that the else value is always discarded and never used. I tried to use a comma instead, this would have been more logical but the compiler then warns that the left hand of the comma is never used. I guess one idiom is just more popular than the other, that's why gcc warns about one but not the other.

I did not test Zephyr but `git grep` does not find any `compile_check` there.